### ISSUE TITLE
Failure to apply inverse patchset

### ISSUE THREAD

**Issue by michaelhabibi** [unadlib/mutative#59]
I've run into a case where `apply` fails when applying the inverse patchset generated by `create`. Take the following code example

```
const parentId = "parent";
const childId = "child";

const myObj = {
  [parentId]: { name: "parent", children: [childId] },
  [childId]: { name: "child" },
};

const [newState, patchset, inverse] = create(
  myObj,
  (draft) => {
    // delete children
    while (draft[parentId].children.length) {
      const id = draft[parentId].children[0];
      draft[parentId].children.splice(0, 1);
      delete draft[id]; // delete child object
    }

    // delete parent
    delete draft[parentId];
  },
  { enablePatches: { arrayLengthAssignment: false } }
);

const reverted = apply(newState, inverse); // throws an error
```

When trying to apply `inverse` to `newState`, `apply` throws the following error:

```
Cannot apply patch at 'parent/children/0'
```

**Expected**
The inverse patchset is correctly generated and applied.

**Note**
Reversing the `inverse` patchset in this case correctly applies it.

**Comment by unadlib** [comment]
hi @michaelhabibi , thanks for reporting the issue. I will fix it later.

**Comment by unadlib** [comment]
Thank you again for reporting this issue.

We're pleased to inform you that the fix for this issue has been implemented and is now available in version v1.0.10, which has been released.

**Comment by michaelhabibi** [comment]
Thank you for such a quick turnaround! Confirming the fix works 👍
